datafusion/sqllogictest/src/engines/datafusion_engine/error.rs

```diff
@@ -30,7 +30,7 @@ pub enum DFSqlLogicTestError {
     #[error("SqlLogicTest error(from sqllogictest-rs crate): {0}")]
     SqlLogicTest(#[from] TestError),
     /// Error from datafusion
-    #[error("DataFusion error: {0}")]
+    #[error("DataFusion error: {}", .0.strip_backtrace())]
```

Comment:
Okay so it turns out that the issue was that the expected error doesn't contain the backtrace but the actual one does. I couldn't reproduce locally at first because I wasn't using the `backtrace` feature.

I have no idea why running with `--complete` didn't write the backtrace to the `.slt` files, I find no call to `strip_backtrace` and I don't know what in my PR made this necessary. But it works 🤷‍♂️

Reply:
I don't think we should add the backtrace to the errors in general (definitely not in `slt`) as the backtraces would include line numbers / call stacks that would change over time.

Reply:
Absolutely. This PR doesn't change that, the backtraces are in the wrapped SchemaError and PlanError. I had to strip them away, don't know why that wasn't necessary before